### DISCUSSION TITLE
What is the correct way to install OSRM

### DISCUSSION THREAD

**Discussion by washaqq** [Project-OSRM/osrm-backend#6815] [Q&A]
Hello,

i don't know if this is the right place to ask, i want to learn what is the correct way to install OSRM for production, i followed a guide (https://gist.github.com/shafayeatsumit/17366c729ede9cbe36768c8bf4add569) and it worked (Built from source), i had no problem, but i am not sure that it is the way!, looking for an answer, please enlighten me.

Kinds.

**Comment by nilsnolde** [comment]
It's entirely up to you how you install this. Docker is often preferred. You might want to consider using the node bindings for prod. There's tons of issues & info in the wiki, best to spend a few days understanding things.

**Comment by washaqq** [comment]
Hi @nilsnolde,

Thank you for your reply, wiki page is complicated, it does not show step by step installation, but it is fine (i am reading), i will figure it out, so as you are saying, i can use either docker or building from the source, both are fine for production, but i did not get and could not find about node binding? what is it? can you help me to understand?

since i don't have xp on docker, i think, i will stick with building from source, i would like to hear your concerns, if there is any.

Edit: is this what you mean? ; https://gis-ops.com/osrm-nodejs-bindings/ 
Kinds.
